components/org.wso2.carbon.identity.conditional.auth.functions.http/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/http/AbstractHTTPFunction.java

```diff
@@ -56,12 +64,18 @@ public abstract class AbstractHTTPFunction {
     protected static final String TYPE_TEXT_PLAIN = "text/plain";
     private static final char DOMAIN_SEPARATOR = '.';
     private static final String RESPONSE = "response";
+    private static final int HTTP_STATUS_UNAUTHORIZED = 401;
+    private int maxRequestAttemptsForAPIEndpointTimeout = 2;
     private final List<String> allowedDomains;
 
     private CloseableHttpClient client;
 
     public AbstractHTTPFunction() {
 
+        if (StringUtils.isNotBlank(IdentityUtil.getProperty(Constants.HTTP_REQUEST_RETRY_COUNT))) {
+            maxRequestAttemptsForAPIEndpointTimeout = Integer.parseInt
+                    (IdentityUtil.getProperty(Constants.HTTP_REQUEST_RETRY_COUNT));
```

Comment:
It seems like we are reading configs from two different places.

The later code reads as 
`ConfigProvider.getInstance()`

Reply:
Fixed with https://github.com/wso2-extensions/identity-conditional-auth-functions/pull/157/commits/05c4f77e8d18ba911c3b98bc7e18ec3fb73dcd33